backend/de.metas.acct.base/src/main/java-legacy/org/compiere/acct/FactLine.java

```diff
@@ -598,13 +605,25 @@ else if (m_doc.getDescription() != null && m_doc.getDescription().length() > 0)
 		}
 
 		// BPartner
+
 		if (m_docLine != null)
 		{
-			setC_BPartner_ID(m_docLine.getBPartnerId());
+			setC_BPartner_Location_ID(m_docLine.getBPartnerLocationId());
+
+			if (getC_BPartner_ID() <= 0)
+			{
+				setC_BPartner_ID(m_docLine.getBPartnerId());
+			}
 		}
-		if (getC_BPartner_ID() <= 0)
+
+		if (getC_BPartner_Location_ID() <= 0)
 		{
-			setC_BPartner_ID(m_doc.getBPartnerId());
+			setC_BPartner_Location_ID(m_doc.getBPartnerLocationId());
+
+			if (getC_BPartner_ID() <= 0)
+			{
+				setC_BPartner_ID(m_doc.getBPartnerId());
+			}
```

Comment:
the code from 611->626 feels a bit twisted.

Reply:
maybe :) it will anyway get better once I introduce `FactLine.setBPartnerAndLocationId`